instrumentation/opentelemetry-instrumentation-celery/tests/test_metrics.py

```diff
@@ -0,0 +1,109 @@
+import time
+import threading
+from timeit import default_timer
+from typing import Union, Optional
+
+from opentelemetry.sdk.metrics._internal.point import Metric
+from opentelemetry.sdk.metrics.export import (
+    HistogramDataPoint,
+    NumberDataPoint,
+)
+from opentelemetry.test.test_base import TestBase
+from opentelemetry.instrumentation.celery import CeleryInstrumentor
+
+from .celery_test_tasks import task_add, app
+
+
+class TestMetrics(TestBase):
+    def setUp(self):
+        super().setUp()
+        self._worker = app.Worker(app=app, pool="solo", concurrency=1,
+                                  hostname='celery@akochavi')
+        self._thread = threading.Thread(target=self._worker.start)
+        self._thread.daemon = True
+        self._thread.start()
+
+    def tearDown(self):
+        super().tearDown()
+        self._worker.stop()
+        self._thread.join()
+
+    def get_metrics(self):
+        CeleryInstrumentor().instrument()
+        result = task_add.delay(1, 2)
+
+        timeout = time.time() + 60 * 1  # 1 minutes from now
+        while not result.ready():
+            if time.time() > timeout:
+                break
+            time.sleep(0.05)
+        CeleryInstrumentor().uninstrument()
+        resource_metrics = (
+            self.memory_metrics_reader.get_metrics_data().resource_metrics
+        )
+
+        all_metrics = []
+        for metrics in resource_metrics:
+            for scope_metrics in metrics.scope_metrics:
+                all_metrics.extend(scope_metrics.metrics)
+
+        return all_metrics
+
+    def assert_metric_expected(
```

Comment:
Copied from **test_metrics_instrumentation.py**

Reply:
Great!
I'm working on moving this code to the TestBase
so we don't copy these functions in each instrumentation:
https://github.com/open-telemetry/opentelemetry-python/pull/3182